platforms/core-configuration/model-core/src/main/java/org/gradle/api/internal/provider/CollectionSupplier.java

```diff
@@ -18,10 +18,29 @@
 
 import java.util.Collection;
 
+/**
+ * A composable {@link ValueSupplier value supplier} that supplies collection values. Instances of
+ * this type are used as value suppliers for {@link AbstractCollectionProperty}.
+ *
+ * @param <T> the type of elements
+ * @param <C> type of collection
+ */
 interface CollectionSupplier<T, C extends Collection<? extends T>> extends ValueSupplier {
+    /**
+     * Realizes and returns the (collection) value.
+     */
     Value<? extends C> calculateValue(ValueConsumer consumer);
 
-    CollectionSupplier<T, C> plus(Collector<T> collector);
+    /**
+     * Composes the current collection supplier with the given {@link Collector} (also a {@link ValueSupplier value supplier}), producing a new
+     * collection supplier that produces a collection that contains the original elements and the
+     * elements supplied by the given collector.
+     *
+     * @param increment a collector that represents an increment to the collection to be returned by this supplier
+     * @return a new supplier that produces a collection that contains the
+     * same elements as this supplier, plus the elements obtained via the given increment collector
+     */
+    CollectionSupplier<T, C> plus(Collector<T> increment);
```

Comment:
💅 `increment` is a somewhat unusual name for this parameter.  Along with the name `plus` for the method itself, it makes me think of arithmetic, instead of insertion into a collection.

Reply:
It is a new "addition", which could be a new element (or provider thereof), or a new batch of elements, or a provider thereof.